<launch>
  <arg name="PL_HOST" default="localhost"/> <!-- SWI-PROLOG only takes requests from localhost --> 
  <arg name="PL_PORT" default="5000"/>
  <arg name="PL_PASSWORD" default="xyzzy"/>
  <arg name="PL_LOGFILE" default="prolog_http_server.log"/>

  <node name="prolog_http_server" pkg="qsr_kb" type="startserver.sh" args="$(arg PL_PORT) $(arg PL_PASSWORD) $(arg PL_LOGFILE)" output="screen"/>

  <node name="kb_server" pkg="qsr_kb" type="kb.py" output="screen">
    <param name="PL_HOST" value="$(arg PL_HOST)"/>
    <param name="PL_PORT" value="$(arg PL_PORT)"/>
    <param name="PL_PASSWORD" value="$(arg PL_PASSWORD)"/>
  </node>  
</launch>
